src/main/webapp/js/query.js

```diff
@@ -177,46 +177,272 @@ function toggleAccountMenu() {
+                restaurantContainerEl.remove();
+                let submitButtonEl = document.getElementById("submit-button");
+                if (submitButtonEl != null) {
+                    submitButtonEl.remove();
+                }
+                modal.style.display = "none";
+            }
+        }
+        feedbackButton = document.createElement('button');
+        feedbackButton.className = 'btn1 feedback';
+        feedbackButton.innerText = "Submit Feedback";
+        feedbackButton.addEventListener('click', () => {
+            let restaurantNameEl = createRestaurantElement(search.name);
+            formEl.appendChild(restaurantNameEl);
+            modal.style.display = "block";
         });
+        newCardBody.appendChild(feedbackButton);
+    }
+    let searchButton = document.createElement('button');
+    searchButton.className = 'btn1 search';
+    searchButton.innerText = "Search with These Parameters Again";
+    newCardBody.appendChild(searchButton);
+    searchButton.addEventListener('click', () => {
+        reroll()});
+    return newCardBody;
+}
+
+//Function to append restaurant name to modal form to force it to follow through to feedback
+function createRestaurantElement(restaurantName) {
+    let userID = 0;
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    let userEl = document.createElement('input');
+    userEl.className = "input--style-2";
+    userEl.type = "text";
+    userEl.id = "user-id";
+    userEl.name = "user-id";
+    userEl.value = userID;
+    userEl.hidden = true;
+
+    let inputGroupEl = document.createElement('div');
+    inputGroupEl.className = "input-group";
+    inputGroupEl.id = "restaurant-name-container";
+    let inputContainer = document.createElement('input');
+    inputContainer.className = "input--style-2";
+    inputContainer.type = "text";
+    inputContainer.id = "restaurant-name-fill";
+    inputContainer.name = "restaurant-name-fill";
+    inputContainer.value = restaurantName;
+    inputContainer.innerText = restaurantName;
+    inputGroupEl.appendChild(inputContainer);
+    inputGroupEl.appendChild(userEl);
+
+    let submitEl = document.createElement('input');
+    submitEl.type = "submit";
+    submitEl.id = "submit-button";
+    inputGroupEl.appendChild(submitEl);
+    return inputGroupEl;
+}
+
+async function fetchFeedback() {
+    let userID = 0;
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    let response = await fetch(`/feedback?user=${userID}`, {
+        method: 'GET'
+    })
+    .then(response => response.json())
+    .then(data => {
+        return data;
+    });
+    return response;
+}
+
+async function getFeedback(search) {
+    let buttons = true;
+    let tempFeedbackElement = "Feedback: You haven't submitted feedback yet";
+    let fetchedFeedback = await fetchFeedback();
+    fetchedFeedback.forEach((feedback) => {
```

Comment:
the response's feedback is a list. is it possible to get the restaurant with restaurantName == search.name without having to loop through everything? It seems like that's the goal of this for loop, but it continues to loop through even if the match has already been found

Reply:
I added a return statement at the end of the if statement so it should end there now

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

> There is no way to stop or break a forEach() loop other than by throwing an exception. If you need such behavior, the forEach() method is the wrong tool.

you would use find instead https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find